wgpu-hal/src/vulkan/device.rs

```diff
@@ -1095,59 +1007,54 @@ impl crate::Device for super::Device {
-            flags.set(
-                gpu_alloc::UsageFlags::DOWNLOAD,
-                desc.usage.contains(wgt::BufferUses::MAP_READ),
-            );
-            flags.set(
-                gpu_alloc::UsageFlags::UPLOAD,
-                desc.usage.contains(wgt::BufferUses::MAP_WRITE),
-            );
-            flags
-        } else {
-            gpu_alloc::UsageFlags::FAST_DEVICE_ACCESS
+        let requirements = unsafe { self.shared.raw.get_buffer_memory_requirements(raw) };
+
+        let is_cpu_read = desc.usage.contains(wgt::BufferUses::MAP_READ);
+        let is_cpu_write = desc.usage.contains(wgt::BufferUses::MAP_WRITE);
+
+        let location = match (is_cpu_read, is_cpu_write) {
+            (true, true) => gpu_allocator::MemoryLocation::CpuToGpu,
+            (true, false) => gpu_allocator::MemoryLocation::GpuToCpu,
+            (false, true) => gpu_allocator::MemoryLocation::CpuToGpu,
+            (false, false) => gpu_allocator::MemoryLocation::GpuOnly,
         };
-        alloc_usage.set(
-            gpu_alloc::UsageFlags::TRANSIENT,
-            desc.memory_flags.contains(crate::MemoryFlags::TRANSIENT),
-        );
 
-        let needs_host_access = alloc_usage.contains(gpu_alloc::UsageFlags::HOST_ACCESS);
+        let needs_host_access = is_cpu_read || is_cpu_write;
 
-        self.error_if_would_oom_on_resource_allocation(needs_host_access, req.size)
+        self.error_if_would_oom_on_resource_allocation(needs_host_access, requirements.size)
             .inspect_err(|_| {
                 unsafe { self.shared.raw.destroy_buffer(raw, None) };
             })?;
 
-        let alignment_mask = req.alignment - 1;
+        let name = desc.label.unwrap_or("Unlabeled buffer");
 
-        let block = unsafe {
-            self.mem_allocator.lock().alloc(
-                &*self.shared,
-                gpu_alloc::Request {
-                    size: req.size,
-                    align_mask: alignment_mask,
-                    usage: alloc_usage,
-                    memory_types: req.memory_type_bits & self.valid_ash_memory_types,
+        let mut allocation = self
+            .mem_allocator
+            .lock()
+            .allocate(&gpu_allocator::vulkan::AllocationCreateDesc {
+                name,
+                requirements: vk::MemoryRequirements {
+                    memory_type_bits: requirements.memory_type_bits & self.valid_ash_memory_types,
+                    ..requirements
                 },
-            )
+                location,
+                linear: true, // Buffers are always linear
+                allocation_scheme: gpu_allocator::vulkan::AllocationScheme::GpuAllocatorManaged,
+            })
+            .inspect_err(|_| {
+                unsafe { self.shared.raw.destroy_buffer(raw, None) };
+            })?;
+
+        // FIXME: Remove this
+        if let Some(slice) = allocation.mapped_slice_mut() {
+            slice.fill(0);
```

Comment:
Currently without this, some `webgpu:api,operation,command_buffer,copyTextureToTexture:copy_depth_stencil:format="stencil8"` [subcases fails](https://github.com/DeltaEvo/wgpu/actions/runs/16834892534/job/47691784286), it seems to tell me that there is either an issue with the BufferInitTracker machinery, or `vkMapMemory` has a side effect that I don't understand